dart/lib/src/utils/sample_rate_format.dart

```
@@ -0,0 +1,289 @@
+      // Overflow into the main digits: 0.99 => 1.00
+      integerPart++;
+      remainingDigits -= digitMultiplier;
+    } else if (_numberOfIntegerDigits(remainingDigits) >
+        _numberOfIntegerDigits(_floor(fraction * digitMultiplier).toInt())) {
+      // Fraction has been rounded (0.0996 -> 0.1).
+      fraction = remainingDigits / digitMultiplier;
+    }
+
+    // Separate out the extra integer parts from the fraction part.
+    extraIntegerDigits = remainingDigits ~/ power;
+    fractionPart = remainingDigits % power;
+
+    var integerDigits = _integerDigits(integerPart, extraIntegerDigits);
+    var digitLength = integerDigits.length;
+    var fractionPresent =
+        fractionDigits > 0 && (minFractionDigits > 0 || fractionPart > 0);
+
+    if (_hasIntegerDigits(integerDigits)) {
+      // Add the padding digits to the regular digits so that we get grouping.
+      var padding = '0' * (_minimumIntegerDigits - digitLength);
+      integerDigits = '$padding$integerDigits';
+      digitLength = integerDigits.length;
+      for (var i = 0; i < digitLength; i++) {
+        _addDigit(integerDigits.codeUnitAt(i));
+      }
+    } else if (!fractionPresent) {
+      // If neither fraction nor integer part exists, just print zero.
+      _addZero();
+    }
+
+    _decimalSeparator(fractionPresent);
+    if (fractionPresent) {
+      _formatFractionPart((fractionPart + power).toString(), minFractionDigits);
+    }
+  }
+
+  /// Helper to get the floor of a number which might not be num. This should
+  /// only ever be called with an argument which is positive, or whose abs()
+  ///  is negative. The second case is the maximum negative value on a
+  ///  fixed-length integer. Since they are integers, they are also their own
+  ///  floor.
+  dynamic _floor(dynamic number) {
+    if (number.isNegative && !number.abs().isNegative) {
+      throw ArgumentError(
+          'Internal error: expected positive number, got $number');
```

Either we do a try-catch when calling SampleRateFormat (and log it) or we swallow this exception returning null.

---

Opted to swallow and return '0' for the sample rate, as i didn't want to make the API nullable. I don't think we will ever experience this code path, as we are only allowing values between 0 and 1.